pkg/pipeline/benchmark.go

```diff
@@ -0,0 +1,193 @@
+package pipeline
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/cncf-tags/green-reviews-tooling/internal/dagger"
+	"github.com/cncf-tags/green-reviews-tooling/pkg/cmd"
+)
+
+const (
+	benchmarkNamespace = "benchmark"
+	falcoNamespace     = "falco" // TODO Remove when no longer used.
+)
+
+type Pipeline struct {
+	container  *dagger.Container
+	dir        *dagger.Directory
+	kubeconfig *dagger.File
+}
+
+func New(container *dagger.Container, dir *dagger.Directory, kubeconfig *dagger.File) (*Pipeline, error) {
+	return &Pipeline{
+		container:  container,
+		dir:        dir,
+		kubeconfig: kubeconfig,
+	}, nil
+}
+
+// Benchmark measures the sustainability footprint of CNCF projects.
+func (p *Pipeline) Benchmark(ctx context.Context,
+	cncfProject,
+	config,
+	version,
+	benchmarkJobURL string,
+	benchmarkJobDurationMins int) (*dagger.Container, error) {
+	_, err := p.benchmark(ctx, cncfProject, config, version, benchmarkJobURL, benchmarkJobDurationMins)
+	if err != nil {
+		log.Printf("benchmark failed: %v", err)
+	}
+
+	_, err = p.delete(ctx, cncfProject, config, benchmarkJobURL)
+	if err != nil {
+		return nil, err
+	}
+
+	return p.container, nil
+}
+
+func (p *Pipeline) Terminal(ctx context.Context) (*dagger.Container, error) {
+	return p.withKubeconfig().Terminal(), nil
+}
+
+func (p *Pipeline) benchmark(ctx context.Context,
+	cncfProject,
+	config,
+	version,
+	benchmarkJobURL string,
+	benchmarkJobDurationMins int) (*dagger.Container, error) {
+	// Create CNCF project resources.
+	_, err := p.deploy(ctx, cncfProject, config, version)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create benchmark job resources.
+	_, err = p.exec(ctx, cmd.Apply(benchmarkJobURL))
+	if err != nil {
+		return nil, err
+	}
+
+	// Wait for pods to be ready.
```

Comment:
idk if the naming is a bit confusing. wait for pods then call wait for namespace. its fine I guess.